pkg/apis/pipeline/v1alpha1/git_resource.go

```diff
@@ -91,26 +91,22 @@ func (s GitResource) GetParams() []Param { return []Param{} }
 
 // Replacements is used for template replacement on a GitResource inside of a Taskrun.
 func (s *GitResource) Replacements() map[string]string {
+
 	return map[string]string{
 		"name":     s.Name,
 		"type":     string(s.Type),
 		"url":      s.URL,
 		"revision": s.Revision,
+		"path":     s.TargetPath,
 	}
 }
 
 func (s *GitResource) GetDownloadContainerSpec() ([]corev1.Container, error) {
 	args := []string{"-url", s.URL,
 		"-revision", s.Revision,
 	}
-	var dPath string
-	if s.TargetPath != "" {
-		dPath = s.TargetPath
-	} else {
-		dPath = s.Name
```

Comment:
hm, is it okay that we no longer default to `s.Name`? (i think maybe when we were pairing we found that `s.Name` was actually a bug? i can't quite remember 😅 )

Reply:
I think we should be defaulting to the resource name (I'll validate!) but the destination directory (or `TargetPath` as it is called in the git resource) is set by the `SetDestinationDirectory` instead of in `NewGitResource` or `GetDownloadContainerSpec`. 

`SetDestinationDirectory` itself is called in [AddInputResources](https://github.com/tektoncd/pipeline/blob/cde3a30c7229f82ed0ef5c39c0ceed49998197ed/pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go#L88) and the value of the path is set in [`destinationPath`](https://github.com/tektoncd/pipeline/blob/cde3a30c7229f82ed0ef5c39c0ceed49998197ed/pkg/reconciler/v1alpha1/taskrun/resources/input_resources.go#L197) which defaults to the name.

This flow is confusing though. I think we see if we can set the destinationDirectory when we create the object instead of the setter.